src/jrd/SystemTriggers.epp

```diff
@@ -0,0 +1,1383 @@
+/*
+ *  The contents of this file are subject to the Initial
+ *  Developer's Public License Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the
+ *  License. You may obtain a copy of the License at
+ *  http://www.ibphoenix.com/main.nfs?a=ibphoenix&page=ibp_idpl.
+ *
+ *  Software distributed under the License is distributed AS IS,
+ *  WITHOUT WARRANTY OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing rights
+ *  and limitations under the License.
+ *
+ *  The Original Code was created by Adriano dos Santos Fernandes
+ *  for the Firebird Open Source RDBMS project.
+ *
+ *  Copyright (c) 2024 Adriano dos Santos Fernandes <adrianosf@gmail.com>
+ *  and all contributors signed below.
+ *
+ *  All Rights Reserved.
+ *  Contributor(s): ______________________________________.
+ */
+
+#include "firebird.h"
+#include <optional>
+#include "firebird/Interface.h"
+#include "../jrd/SystemTriggers.h"
+#include "../jrd/constants.h"
+#include "../jrd/ids.h"
+#include "../jrd/ini.h"
+#include "../jrd/req.h"
+#include "../jrd/dpm_proto.h"
+#include "../jrd/err_proto.h"
+#include "../jrd/exe_proto.h"
+#include "../jrd/evl_proto.h"
+#include "../jrd/met_proto.h"
+#include "../jrd/mov_proto.h"
+
+
+DATABASE DB = FILENAME "ODS.RDB";
+
+using namespace Firebird;
+using namespace Jrd;
+
+
+namespace
+{
+
+void beforeDeleteCheckConstraint(thread_db* tdbb, Record* record)
```

Comment:
It is better to name routines according to what they do instead of when they are called.